net/tstun/wrap.go

```diff
@@ -955,39 +951,116 @@ func (t *Wrapper) Read(buffs [][]byte, sizes []int, offset int) (int, error) {
 	return buffsPos, res.err
 }
 
-// injectedRead handles injected reads, which bypass filters.
-func (t *Wrapper) injectedRead(res tunInjectedRead, buf []byte, offset int) (int, error) {
-	metricPacketOut.Add(1)
+const (
+	minTCPHeaderSize = 20
+)
 
-	var n int
-	if !res.packet.IsNil() {
+func stackGSOToTunGSO(pkt []byte, gso stack.GSO) (tun.GSOOptions, error) {
+	var gsoType tun.GSOType
+	needTCPhlen := true
+	var tcphLen uint16
+	switch gso.Type {
+	case stack.GSONone:
+		gsoType = tun.GSONone
+		needTCPhlen = false
```

Comment:
If `options` is initialized above this `switch`, then rather than a flag, this could be an early return.

Reply:
Good point. Updated in https://github.com/tailscale/tailscale/pull/12869/commits/87089f8c15bc139e63bb7415deed87100434f95d